src/core/thread/mle.cpp

```diff
@@ -3588,6 +3604,11 @@ otError Mle::HandleChildUpdateResponse(const Message &         aMessage,
     switch (mRole)
     {
     case OT_DEVICE_ROLE_DETACHED:
+        // Version
+        SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kVersion, sizeof(version), version));
+        VerifyOrExit(version.IsValid(), error = OT_ERROR_PARSE);
+        mParent.SetVersion(static_cast<uint8_t>(version.GetVersion()));
```

Comment:
I am not sure but I think this may break compatibility in behavior between different OT builds (before and after this change)
- Here we expect in `HandleChildUpdateResponse()` that the message should include `VersionTlv`.
- On parent side the code that appends this TLV is added in this PR itself (from the change in `MleRouter::SendChildUpdateResponse()`)
- If we have a parent that is running a version of OT before this change, the child would reject its "Child Update Response".

Reply:
The other places we add a check for VersionTlv in this PR are in `Mle::HandleParentResponse()` and `MleRouter::HandleChildIdRequest()`. Checking the src I see that in current OT master we do already include the version TLV in sent "parent response" or "child id request" msgs, so adding the check in those two should be fine.

Reply:
Thread 1.2.0 Specification does not currently indicate the use of Version TLV. Rather than querying the parent's version, how about saving the parent's version as as part of `ParentInfo` in settings (i.e. non-volatile). Alternatively, we'd need to propose an update to the Thread Specification.

Reply:
@abtink This PR explicitly requests the version TLV when sending child update request. I don't think this will break compatibility. We may choose to save the version, but requesting version in child update request should also work. Thoughts @jwhui

Reply:
Current implementation will never include a Version TLV in the Child Update Response. However, this PR currently requires Version TLV to be included in the Child Update Response.

Reply:
Oh, yes. I will make this optional.

Reply:
I think if we want to allow the possibility of requesting Version TLV, we need to make that more explicit in the Thread 1.2.0 Specification. I submitted [SPEC-909](https://threadgroup.atlassian.net/browse/SPEC-909) for further discussion.

Reply:
I think best approach for now is to include in request but make parsing of it optional. 
- We include in list of of requested TLVs we sent out in "Child Update Requset"
- Parent running 1.1 version (earlier OT build) would not include this TLV in the "Child Update Response"
- Here on the child, we check if Version TLV is present in the message, we parse it. Otherwise   we assume parent is 1.1.

We can also save version in `ParentInfo` (during parsing we can consider it 1.1 if it is not present), but I favor doing just an optional parsing of the TLV during the "Child Update Response". 


Thoughts?

Reply:
@abtink, I’m fine with the approach, but we need to ensure the Thread Specification explicitly supports querying the Version TLV (SPEC-909 was raised with the Thread Group).

Reply:
Updated to persist parent version and process ChildUpdateResponse as proposed by @abtink

Reply:
To add the ability to request Version TLV, we'll need to wait on the outcome of [SPEC-909](https://threadgroup.atlassian.net/browse/SPEC-909). This also raises another issue of what to do if a device upgrades from Thread 1.1 to Thread 1.2, since Thread 1.1 devices may/will not store Version of peers persistently. I have filed [SPEC-911](https://threadgroup.atlassian.net/browse/SPEC-911) to discuss this issue.

Reply:
I removed requesting Version TLV in Child Update Request.